### PR TITLE
Achievement Get Part 3 - Cargo Achievements (and bugfixes!)

### PR DESCRIPTION
**Pull request by Altoids1**
image

This adds four more achievements, as part of #7657, of the Cargonian variety.

## Coder Warning
For some reason, #7751 made both read and write of the database entries for achievements no longer work.

I have... no real idea why, but I've decided to revert Alexkar's changes and do the sanitation through new methods.

All achievements now use the ``initial()`` variants of their properties to prevent tampering.

In addition, all ckeys used in queries are run through ``ckey()``, which would remove any and all special characters that're necessary to do a SQL injection.

Doing it this way should not only be more performative (using native BYOND procs instead of several sequential calls to bsql.dll) but also actually work.

#### Changelog
:cl:  Altoids
rscadd: Cargo now has a couple achievements of their own!
bugfix: Fixed bug that caused achievements to not load nor save.
/:cl:

### CODE REVIEW

**Comment by boodaliboo** [comment]
Assistant

**Comment by Altoids1** [comment]
# Ready for review.

**Review by Hopekz — APPROVED** [review]

File: code/modules/economy/account.dm [hunk 1]
```diff
@@ -30,6 +32,14 @@
...
 	if(account_balance < 0)
 		account_balance = 0
+	else if(account_balance > 1000000 && !is_bourgeois) // if we are now a millionaire, give the achievement
+		//So we currently only know what is *supposed* to be the real_name of the client's mob. If we can find them, we can get them this achievement. 
+		for(var/x in GLOB.player_list)
+			var/mob/M = x
+			if(M.real_name == account_holder)
+				SSachievements.unlock_achievement(/datum/achievement/cargo/bourgeois, M.client)
+				is_bourgeois = TRUE
+				break
```
Comment (alexkar598) [inline code/modules/economy/account.dm]: You can't assume there aren't going to be any duplicated names due to genetic fuckery or something like that
```suggestion
```

File: code/modules/awaymissions/bluespaceartillery.dm [hunk 2]
```diff
@@ -47,6 +47,8 @@
 	if(usr.contents.Find(src) || (in_range(src, usr) && isturf(loc)) || issilicon(usr))
 		priority_announce("Bluespace artillery fire detected. Brace for impact.")
 		message_admins("[ADMIN_LOOKUPFLW(usr)] has launched an artillery strike.")
+		if(usr.client)
+			SSachievements.unlock_achievement(/datum/achievement/cargo/bsa, usr.client)
```
Comment (alexkar598) [inline code/modules/awaymissions/bluespaceartillery.dm]: This is an admin only version, not the cargo not
```suggestion
		message_admins("[ADMIN_LOOKUPFLW(usr)] has launched an artillery strike.")
```

File: code/modules/economy/account.dm [hunk 3]
```diff
@@ -30,6 +32,14 @@
 	account_balance += amt
 	if(account_balance < 0)
 		account_balance = 0
+	else if(account_balance > 1000000 && !is_bourgeois) // if we are now a millionaire, give the achievement
+		//So we currently only know what is *supposed* to be the real_name of the client's mob. If we can find them, we can get them this achievement. 
+		for(var/x in GLOB.player_list)
+			var/mob/M = x
+			if(M.real_name == account_holder)
+				SSachievements.unlock_achievement(/datum/achievement/cargo/bourgeois, M.client)
```
Comment (alexkar598) [inline code/modules/economy/account.dm:40]: Please add a check within the unlock_achievement proc to check if the client exists, its too easy to forget a ``if(whatever.client)``

**Review by alexkar598 — CHANGES_REQUESTED** [review]

**Pull request merged**